lib/AST/IRToASTVisitor.cpp

```diff
@@ -691,29 +686,19 @@ clang::Stmt *IRToASTVisitor::visitGetElementPtrInst(
     }
   }
 
-  return GetTempAssign(inst, ast.CreateAddrOf(base));
+  return ast.CreateAddrOf(base);
 }
 
-clang::Stmt *IRToASTVisitor::visitInstruction(llvm::Instruction &inst) {
+clang::Expr *IRToASTVisitor::visitInstruction(llvm::Instruction &inst) {
   THROW() << "Instruction not supported: " << LLVMThingToString(&inst);
   return nullptr;
 }
 
-clang::Stmt *IRToASTVisitor::visitBranchInst(llvm::BranchInst &inst) {
-  DLOG(INFO) << "visitBranchInst ignored: " << LLVMThingToString(&inst);
-  return ast.CreateNullStmt();
-}
-
-clang::Stmt *IRToASTVisitor::visitUnreachableInst(llvm::UnreachableInst &inst) {
-  DLOG(INFO) << "visitUnreachableInst ignored:" << LLVMThingToString(&inst);
-  return ast.CreateNullStmt();
-}
-
-clang::Stmt *IRToASTVisitor::visitExtractValueInst(
+clang::Expr *IRToASTVisitor::visitExtractValueInst(
     llvm::ExtractValueInst &inst) {
   DLOG(INFO) << "visitExtractValueInst: " << LLVMThingToString(&inst);
 
-  auto base{GetOperandExpr(inst.getAggregateOperand())};
+  auto base{GetOperandExpr(inst.getOperandUse(0))};
```

Comment:
Is there some other way to get the `llvm::Use` that would be more easily understandable? The indexes are kind of opaque to me personally.

Reply:
Not in general case I'm afraid... Some instructions expose a method to get the index, some of them only as `protected`, others not at all